File: java/002/Main.java
/*
 * ProjectEuler 002
 * Java / Main.java
 * copyright (c) 2015 Susisu
 *
 * Java SE 1.8.0_25
 * `javac Main.java -d bin`
 */

public class Main {
    public static void main(String[] args) {
        int sum = 0;
        int a = 0;
        int b = 1;
        while (a <= 4000000) {
            int t = a;
            a     = b;
            b    += t;
            if (a % 2 == 0) {
                sum += a;
            }
        }
        System.out.println(sum);
    }
}

File: ocaml/002/main.ml
(*
    ProjectEuler 002
    OCaml / main.ml
    copyright (c) 2015 Susisu

    OCaml 4.02.1
    `ocamlc main.ml -o bin/main`
*)

let rec calc a b max sum =
    if a <= max then
        if a mod 2 = 0 then
            calc b (a + b) max (sum + a)
        else
            calc b (a + b) max sum
    else
        sum;;

let main () =
    begin
        print_int (calc 1 1 4000000 0);
        print_newline ()
    end;;

main ();;
